importing/serializers.py

```diff
@@ -0,0 +1,70 @@
+from rest_framework import serializers
+
+from formatting.models import Format
+from station.models import Station
+
+from .models import DataImport
+
+
+class DataImportUploadRequestSerializer(serializers.Serializer):
+    """Serializer for data import upload request."""
+
+    station = serializers.SlugRelatedField(
+        slug_field="station_code",
+        queryset=Station.objects.all(),
+        help_text="Station code",
+    )
+    format = serializers.PrimaryKeyRelatedField(
+        queryset=Format.objects.all(),
+        help_text="Format ID",
+    )
+    visibility = serializers.ChoiceField(
+        choices=["public", "private"],
+        default="private",
+        help_text="Visibility level",
+    )
+    reprocess = serializers.BooleanField(
+        default=False,
+        help_text="Reprocess data after import",
+    )
+    observations = serializers.CharField(
+        required=False,
+        allow_blank=True,
+        help_text="Additional observations",
+    )
+    rawfile = serializers.FileField(
+        help_text="Data file to upload",
+    )
```

Comment:
I think that, in this case, you should be able to use a `ModelSerializer`, as you do below, just with other fields:


```suggestion
class DataImportUploadRequestSerializer(serializers.ModelSerializer):
    """Serializer for data import upload request."""

    class Meta:
        model = DataImport
        fields = ["station", "format", "visibility", "reprocess", "observations", "rawfile"]
```

Reply:
@dalonsoa I would keep this as is for now, since we want to specify the `station_code` rather than the `station_id`. I am happy to change it if you would rather specify the ID